src/utils/template_handler.rs

```diff
@@ -129,24 +149,39 @@ pub(crate) fn load_remote_template_collection(path: &str, url: &str, force: bool
                     .replace('"', "")
                     .as_str(),
                 force,
+                Some(&url_type),
             );
             if !st.is_ok {
                 return st;
             }
         }
     }
+
     Status::ok()
 }
 
 /// Load a template from a remote repository
-pub(crate) fn load_remote_template(path: &str, url: &str, force: bool) -> Status {
+pub(crate) fn load_remote_template(
+    path: &str,
+    url: &str,
+    force: bool,
+    url_type: Option<&URLType>,
```

Comment:
I kept url_type as optional here because at some places this function is called where url_type is already calculated, but in some url_type is not there, so in order to provide more flexiibilty, I made this as optional, so if url_type is calculated before and is passed into this function we will take thaaat otherwise will recalculate  later in this code ,